### ISSUE TITLE
Not working

### ISSUE THREAD

**Issue by Perkin5** [evgeny-golubev/Laravel-4-Artisan#5]
I have Sublime Text 2 with the Laravel 4 Artisan and Generator packages installed. I want to generate a controller.
I hit Ctrl-Shift-P and type Laravel and get the dropdown with the following options:
Laravel Artisan Command and Laravel Generate: Command, Migration, Model, Resource, Scaffold, Seed, Test, View
I notice there is no Generate:Controller option so I click Laravel Artisan Command and then type into the console: Controller:make TestController, press Enter and nothing happens.
Any ideas?

**Comment by evgeny-golubev** [comment]
Any messages in console CTRL+~?
Are you insert in Sublime Text user settings "show_panel_on_build": true or use Tools/Build Results/Show Build Results menu item for view results?

**Comment by Perkin5** [comment]
Console message:

Traceback (most recent call last):
  File ".\generate.py", line 88, in call_artisan
  File ".\subprocess.py", line 633, in **init**
  File ".\subprocess.py", line 842, in _execute_child
WindowsError: [Error 2] The system cannot find the file specified

Tools/Build Results/Show Build Results is grayed out

**Comment by evgeny-golubev** [comment]
By default in the settings `"php_path": "php"`. You can change it to your path in `Preferences\Package Settings\LaravelArtisan\Settings – User`. 

You have to use next code:

``` JSON
{
    "php_path": "your path to php"
}
```

Maybe the problem is that.

**Comment by Perkin5** [comment]
Put in following: <code> {
    "php_path": "C:\wamp\bin\php\php5.3.8\php.exe"
} </code> and restarted - which resulted in no response at all after entering artisan command, not even a console message. Did I enter the path in the right format? I noticed incidentally that the default settings file is blank - should it be?

**Comment by evgeny-golubev** [comment]
What command you use in console for executing artisan without plugin?

Are you insert in Sublime Text user settings "show_panel_on_build": true or use Tools/Build Results/Show Build Results menu item for view results?

Default user settings file is blank on install.

**Comment by Perkin5** [comment]
Sorry, closed by mistake. The artisan command I am using is 
<code>controller:make TestController</code>
and the console message is still
<code>
Traceback (most recent call last):
  File ".\generate.py", line 88, in call_artisan
  File ".\subprocess.py", line 633, in **init**
  File ".\subprocess.py", line 842, in _execute_child
WindowsError: [Error 2] The system cannot find the file specified
</code>
Sublime Text user settings do already contain the item "show_panel_on_build":true
The Tools/Build Results/Show Build Results menu is grayed out.

**Comment by evgeny-golubev** [comment]
I need answer for question: `What command you use in console for executing artisan without plugin?`.

**Comment by Perkin5** [comment]
Don't understand question - without what plugin? I press Ctrl-Shift-P, type in 'artisan', click Laravel Artisan Command and in the 'Enter an artisan command' field at the bottom of the window I type:
controller:make TestController
Is that the answer you need?

**Comment by evgeny-golubev** [comment]
Ok. I don't use Windows and you need test composer in your terminal before i can help you.
1) Go to this section: http://getcomposer.org/doc/00-intro.md#installation-windows
2) Test in your current terminal execution `composer -V`. When you get the correct result, write here the full command for correct use composer in your system.

**Comment by Perkin5** [comment]
I was confident that composer was working properly as I use it to install Laravel 4 into a project.
I type the following into the terminal:
<code>composer create-project laravel/laravel</code>
I typed 'composer -v' into the GitBash command line and got this:
<code>Composer version d017e3f2096dda7194600a76bc7328808811c8da</code>
plus a list of options and available commands.
Is that OK?

**Comment by evgeny-golubev** [comment]
Ok.
1) Press `Start+R`.
2) Type `cd \path\to\laravel\project\`.
3) Type `php artisan help`. You should get an error.
4) Type `cmd set PATH=%PATH;path\to\php\folder\`.
5) Type `php artisan help`. You should get artisan help.
6) Change php path in `Preferences\Package Settings\LaravelArtisan\Settings – User` to:

``` JSON
{
    "php_path": "php"
}
```

Read more on Laravel Forums: http://forums.laravel.io/viewtopic.php?id=712

**Comment by Perkin5** [comment]
I got as far as step 3 but then I didn't get an error. I got:
<code>
$ php artisan help
Usage:
 help [--xml] [--format="..."] [--raw] [command_name]

Arguments:
 command               The command to execute
 command_name          The command name (default: "help")

Options:
 --xml                 To output help as XML
 --format              To output help in other formats
 --raw                 To output raw command help
 --help (-h)           Display this help message.
 --quiet (-q)          Do not output any message.
 --verbose (-v|vv|vvv) Increase the verbosity of messages: 1 for normal out
2 for more verbose output and 3 for debug
 --version (-V)        Display this application version.
 --ansi                Force ANSI output.
 --no-ansi             Disable ANSI output.
 --no-interaction (-n) Do not ask any interactive question.
 --env                 The environment the command should run under.

Help:
 The help command displays help for a given command:

   php artisan help list

 You can also output the help in other formats by using the --format option

   php artisan help --format=xml list

 To display the list of available commands, please use the list command.
</code>

which I guess is the correct response?

**Comment by evgeny-golubev** [comment]
Strange.

Change php path in Preferences\Package Settings\LaravelArtisan\Settings – User to:

``` JSON
{
    "php_path": "php"
}
```

**Comment by Perkin5** [comment]
I've done that and then tried again to create the controller but I get the same console message:

<code>Traceback (most recent call last):
  File ".\generate.py", line 88, in call_artisan
  File ".\subprocess.py", line 633, in **init**
  File ".\subprocess.py", line 842, in _execute_child
WindowsError: [Error 2] The system cannot find the file specified</code>

**Comment by evgeny-golubev** [comment]
Ок. I will try package in Windows on Parallels in the evening and return with results.

**Comment by Perkin5** [comment]
Many thanks

**Comment by evgeny-golubev** [comment]
Update and try now. Perhaps the problem is corrected.

**Comment by Perkin5** [comment]
Tried that but result the same. Error is:

Traceback (most recent call last):
   File ".\generate.py", line 96, in call_artisan
   File ".\subprocess.py", line 633, in **init**
   File ".\subprocess.py", line 842, in _execute_child
WindowsError: [Error 2] The system cannot find the file specified

The line numbers have changed but still no joy. Baffled.

Mike

**Comment by evgeny-golubev** [comment]
Try next:

1) The full path (with php.exe) in the end should go in the `php_path`, like this:

``` json
{
    "php_path": "C:\\xampp\\php\\php.exe"
}
```

2) Run ST as Administrator.

**Comment by Perkin5** [comment]
Done that but same result.

Artisan Command line:

controller:make TestController

Path like this:

{
     "php_path": "C:\wamp\bin\php\php5.3.8\php.exe"
}

Error:

Traceback (most recent call last):
   File ".\generate.py", line 96, in call_artisan
   File ".\subprocess.py", line 633, in **init**
   File ".\subprocess.py", line 842, in _execute_child
WindowsError: [Error 2] The system cannot find the file specified

**Comment by evgeny-golubev** [comment]
You ran ST as Administrator?

**Comment by Perkin5** [comment]
I sure did.

**Comment by Perkin5** [comment]
Laravel Artisan still seems not to be working but some new information. 
I press Ctrl-Shift P, type 'artisan' and get the menu of options. Top of the list is Laravel Artisan Command (with no colon between Artisan and Command) and this is what I have been using and failing with. I click it and enter 'controller:make test' and nothing happens. However, further down the menu is an option Laravel Artisan: Controller: Make. If I click this and enter 'test' a restful controller called 'test' is successfully created. How do you think that can be explained?

**Comment by berpcor** [comment]
I haven't got local server installed (mine is portable). But I did everything for to work with php from a command line. 
Composer is working. But... artisan package is not. I noticed one thing in logs:

Running p h p   W : \ d o m a i n s \ l a r a v e l \ a r t i s a n   m i g r a t e
Traceback (most recent call last):
  File ".\sublime_plugin.py", line 337, in run_
    return self.run(**args)
  File ".\exec.py", line 154, in run
  File ".\exec.py", line 45, in **init**
UnicodeDecodeError: 'ascii' codec can't decode byte 0xcf in position 23: ordinal not in range(128)

As you see, there are spaces between all the letters. Is it normal?
